options/db_options.cc

```diff
@@ -1056,6 +1070,34 @@ void MutableDBOptions::Dump(Logger* log) const {
                    compaction_readahead_size);
   ROCKS_LOG_HEADER(log, "                 Options.max_background_flushes: %d",
                           max_background_flushes);
+  ROCKS_LOG_HEADER(log, "Options.daily_offpeak_start_time_utc: %s",
+                   daily_offpeak_start_time_utc.c_str());
+  ROCKS_LOG_HEADER(log, "Options.daily_offpeak_end_time_utc: %s",
+                   daily_offpeak_end_time_utc.c_str());
+}
+
+bool MutableDBOptions::IsNowOffPeak(SystemClock* clock) const {
+  if (daily_offpeak_start_time_utc.empty() ||
+      daily_offpeak_end_time_utc.empty()) {
+    return false;
+  }
+  int64_t now;
+  if (clock->GetCurrentTime(&now).ok()) {
+    time_t now_unix_time = static_cast<time_t>(now);
+    // extract hour and minute from unix_time_now in HH:mm format
+    char now_utc[6];
+    std::strftime(now_utc, sizeof(now_utc), "%R", std::gmtime(&now_unix_time));
```

Comment:
Leaning on standard APIs is generally a good idea, but I worry that we're bringing in extra complexity that we might not fully understand and foresee corner cases of. We compile on a lot of different systems and locales, and we have been hit by variances in std api behavior before.

Note that unix time 0 is midnight UTC and that generalizes such that any unix time % 86400 is the number of seconds since midnight. I think the IsNowOffPeak determination should be made by converting to these "seconds since midnight" values. String comparison for time comparison is somewhat fragile, as it relies on zero padding and (here) implicit conversion of now_utc from char* to string. "Seconds since midnight" makes it easier to expand the format of our allowed time strings in the future, without really affecting this function (assuming we have a parse function used by both this function and ValidateOptions).

Reply:
Yes, I sort of realized that "seconds since midnight" would work after writing the test. Because what I did in the test to set the time was basically `SetCurrentTime(now_utc_hour * 3600 + now_utc_minute * 60)`). Just thinking backwards, this would work for checking the time as well. Let me make the change.